src/Components/Web.JS/src/Platform/Mono/MonoPlatform.ts

```diff
@@ -369,6 +369,10 @@ async function createRuntimeInstance(resourceLoader: WebAssemblyResourceLoader):
   setModuleImports('blazor-internal', {
     Blazor: { _internal: Blazor._internal },
   });
+  const exports = await runtime.getAssemblyExports('Microsoft.AspNetCore.Components.WebAssembly');
+  Object.assign(Blazor._internal, {
+    ...exports.Microsoft.AspNetCore.Components.WebAssembly.Services.DefaultWebAssemblyJSRuntime,
+  });
```

Comment:
This is very interesting. It's certainly nice that we can now add exports to the assembly and they just show up on `Blazor._internal` without further manual work!

One possible drawback is that this is now pretty implicit and magic. If a developer is trying to find the implementation of `Blazor._internal.someMethod`, they may struggle to do so, because literally nothing on the TS side provides an implementation of `someMethod` or even declares the symbol, so no amount of reading the TS sources is going to explain what's going on at runtime. The developer would have to use a debugger and some kind of binary-search is-it-assigned-yet approach to even discover  the magic line 373 and the fact that the implementation comes automatically from .NET code.

If we did want to make this clearer, we could do something like:

 * For the methods that are only used on WebAssembly, put them in some additional `Blazor._internal.dotNetExports.*` grouping. Then the developer would naturally search for `dotNetExports` to see what populates it and would immediately get here.
 * For methods that are not WebAssembly-specific, and hence need to remain at `Blazor._internal.methodName`, we could assign them manually, e.g.: `Blazor._internal.handleNavigation = dotNetExports.handleNavigation`, so again a naive "find-in-files" would bring you here.

I know that's more work to do, and makes literally no difference at runtime. Maybe this isn't really a problem and we should just carry on. It's just that these are the kinds of maintenance friction points I've learned to be suspicious about. I'll leave it with you to decide if you think this is discoverable enough already through some other means!

Reply:
+1 for `dotNetExports` I don't mind `...` expansion after that because `dotNetExports` would make it discoverable (to me)